server/py/services/alerts/main.py

```diff
@@ -451,6 +451,35 @@ async def _filter_alert_activations_by_permissions(_alert_activations):
             pagination=page_info,
         )
 
+    async def get_alert_activation(
+        self,
+        request: fastapi.Request,
+        project: str,
+        name: Optional[str],
+        activation_id: int,
+        auth_info: mlrun.common.schemas.AuthInfo,
+        db_session: sqlalchemy.orm.Session = None,
+    ) -> mlrun.common.schemas.AlertActivation:
+        alert_activation = await run_in_threadpool(
```

Comment:
This should happen after the call to `query_project_resource_permissions` so we first check permissions, then execute the query.